lib/cucumber/runtime.rb

```diff
@@ -257,15 +257,15 @@ def filters
           filters << Filters::ApplyAroundHooks.new(@support_code)
         end
 
-        filters << Filters::BroadcastTestCaseReadyEvent.new(@configuration)
-
         unless configuration.dry_run?
           filters << Filters::BroadcastTestRunStartedEvent.new(@configuration)
           filters << Filters::Quit.new
           filters << Filters::Retry.new(@configuration)
           # need to do this last so it becomes the first test step
           filters << Filters::PrepareWorld.new(self)
         end
+
+        filters << Filters::BroadcastTestCaseReadyEvent.new(@configuration)
```

Comment:
Note to self (cc/ @davidjgoss )

This is the cause of the retry CCK blowing up